packages/interface/src/components/complementary-area/index.js

```diff
@@ -243,10 +243,13 @@ function ComplementaryArea( {
 	useEffect( () => {
 		// Set initial visibility: For large screens, enable if it's active by
 		// default. For small screens, always initially disable.
-		if ( isActiveByDefault && activeArea === undefined && ! isSmall ) {
-			enableComplementaryArea( scope, identifier );
-		} else if ( activeArea === undefined && isSmall ) {
-			disableComplementaryArea( scope, identifier );
+		if ( activeArea === undefined ) {
+			const args = [ scope, identifier, { persist: false } ];
+			if ( isActiveByDefault && ! isSmall ) {
+				enableComplementaryArea( ...args );
+			} else if ( isSmall ) {
+				disableComplementaryArea( ...args );
+			}
 		}
```

Comment:
I saw this is originally from #22381, @jorgefilipecosta

Reply:
Looking at this again, I wonder if it's a good idea for the component to have this kind of logic.

It'd probably simplify things if the component only had to be aware of whether it was active or not, and instead the editor packages controlled showing it by default.

I guess it's a stable API now so it's probably not something that can be simplified.